homeassistant/components/romy/config_flow.py

```diff
@@ -0,0 +1,160 @@
+"""Config flow for ROMY integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import romy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN, LOGGER
+
+
+def _schema_with_host() -> vol.Schema:
+    return vol.Schema(
+        {
+            vol.Required(CONF_HOST): cv.string,
+        },
+    )
+
+
+def _schema_with_password() -> vol.Schema:
+    return vol.Schema(
+        {vol.Required(CONF_PASSWORD): vol.All(cv.string, vol.Length(8))},
+    )
+
+
+class RomyConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle config flow for ROMY."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Handle a config flow for ROMY."""
+        self.host: str = ""
+        self.password: str = ""
+        self.robot_name_given_by_user: str = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            self.host = user_input[CONF_HOST]
+
+            new_romy = await romy.create_romy(self.host, "")
+
+            if not new_romy.is_initialized:
+                errors[CONF_HOST] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(new_romy.unique_id)
+                self._abort_if_unique_id_configured()
+
+                self.robot_name_given_by_user = new_romy.user_name
+
+                if not new_romy.is_unlocked:
+                    return await self.async_step_password()
+
+            if not errors:
+                return await self._async_step_finish_config()
```

Comment:
I believe this is equivalent is more straight forward to reduce the number of branches:

```suggestion
                if not new_romy.is_unlocked:
                    return await self.async_step_password()
                return await self._async_step_finish_config()
```

Reply:
I recommend simplifying the flow in the password step two (after addressing the other comment about not posting back to the first step)

Reply:
in password step the if not erros makes sense cause there are two cases which get handled